\documentclass[12pt]{article}

\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc}
\usepackage{libertine}
\usepackage{libertinust1math}
\usepackage{sourcecodepro}

\usepackage{microtype}

\usepackage[english]{babel}
\usepackage{csquotes}

\usepackage[top=1in, bottom=1.25in, left=1.25in, right=1.25in]{geometry}

\usepackage{booktabs}

\usepackage{graphicx}

\usepackage{solarized-light}
\lstset{%
  basicstyle = \ttfamily\footnotesize, 
  numbers    = left, 
  language   = python,
  aboveskip  = 20pt,
  belowskip  = 20pt
}

\usepackage{hyperref}
\usepackage{xcolor-solarized}

\hypersetup{%
  linkcolor  = solarized-blue,
  citecolor  = solarized-blue,
  urlcolor   = solarized-blue,
  colorlinks = true
}

\renewcommand{\UrlFont}{\normalsize}

% Personal LaTeX styles, see github.com/sdiebolt/latex-styles.
\usepackage{basicmaths}

\begin{document}
  \pagenumbering{Alph}
  \begin{titlepage}
    \begin{center}
      \vspace*{1cm}
 
      \Huge
      \textbf{Systems Biology \& Neurobiology}
 
      \vspace{0.5cm}
      \LARGE
      Homework Report
 
      \vspace{1.5cm}
 
      \Large\textbf{Simon \textsc{Chardin}, Ã‰mile \textsc{Sabatier}, Samuel
      \textsc{Diebolt}}

      \vspace{0.8cm}
 
      %\includegraphics[width=0.7\textwidth]{readdigits}
      \vfill
 
      \vspace{0.8cm}

      \includegraphics[width=0.8\textwidth]{espci_logo}
 
      \Large
      \textbf{Teachers:} Philippe Nghe, Andrew Griffiths
    \end{center}
    \thispagestyle{empty}
  \end{titlepage}
  \pagenumbering{arabic}


  \section*{Abstract}

  \textbf{Deep mutational scanning (DMS) makes use of large-scale mutagenesis
  to reveal intrinsic protein properties, functions and the consequences of
  genetic variation. Recently, the CRISPR/Cas9-mediated genomic error-prone
  editing (CREPE) technology was developed as a high-throughput method for
  mutating essential genes of \textit{Escherichia coli}~\cite{choudhury2020}.
  Its authors applied the technology to target \textit{rpoB}, the gene encoding
  the $\beta$ subunit of bacterial RNA polymerase, and used deep sequencing to
  study resistance against the antibiotic rifampicin. In particular, the
  authors studied epistastis effects by comparing fitness of double mutants in
  \textit{rpoB} with those from the respective single mutations in the presence
  of rifampicin. In this report, we replicated the aforementioned epistasis
  study using a simplified dataset provided by A. Choudhury.}

  \section*{Motivations \& Hypotheses}

  \section*{State of the Art}

  \section*{Methods \& Results of the Supporting Article}

  \section*{Effects of Epistasis in Double Mutants}

  In this section, we replicated the analyses performed by the authors of the
  article supporting this report to better understand the impact of epistasis
  on rifampicin resistance. In particular, we were interested in comparing the
  fitness of double mutants, compared to the sum of fitness from the respective
  single mutations.

  \section*{Next Steps}



  \newpage

  \bibliographystyle{ieeetr}
  \bibliography{main}

\end{document}
